cmd/nerdctl/pull.go

```diff
@@ -92,6 +100,54 @@ func pullAction(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if isVerify, err := cmd.Flags().GetString("verify"); err == nil && isVerify == "cosign" {
+		digest, err := imgutil.ResolveDigest(ctx, rawRef, false)
+		rawRef = rawRef + "@" + digest
+		if err != nil {
+			logrus.Errorf("Unable to resolve digest for an image %s: %v\n", rawRef, err)
+		}
+
+		logrus.Debugf("verifying image: %s\n", rawRef)
+
+		cosignExecutable, err := exec.LookPath("cosign")
+		if err != nil {
+			logrus.WithError(err).Error("cosign executable not found in path $PATH")
+			logrus.Info("you might consider installing cosign from: https://docs.sigstore.dev/cosign/installation")
+			return err
+		}
+
+		cosignCmd := exec.Command(cosignExecutable, []string{"verify"}...)
+		cosignCmd.Env = os.Environ()
+
+		keyRef, err := cmd.Flags().GetString("cosign-key")
+		if err != nil {
+			return err
+		}
+
+		if keyRef != "" {
+			cosignCmd.Args = append(cosignCmd.Args, "--key", keyRef)
+		} else {
+			cosignCmd.Env = append(cosignCmd.Env, "COSIGN_EXPERIMENTAL=true")
```

Comment:
this should be `COSIGN_EXPERIMENTAL=1`  for keyless mode, right ?

Reply:
hello @fahedouch, both cases are valid 🙏